daemon/checkpoint.go

```diff
@@ -56,13 +62,20 @@ func (daemon *Daemon) CheckpointDelete(name string, config types.CheckpointDelet
 	}
 
 	var checkpointDir string
+	var checkpointAbsDir string
+	var out error
 	if config.CheckpointDir != "" {
-		checkpointDir = config.CheckpointDir
+		checkpointDir = filepath.Join(config.CheckpointDir, container.ID, "checkpoints")
 	} else {
 		checkpointDir = container.CheckpointDir()
 	}
-
-	return os.RemoveAll(filepath.Join(checkpointDir, config.CheckpointID))
+	checkpointAbsDir = filepath.Join(checkpointDir, config.CheckpointID)
+	if stat, err := os.Stat(checkpointAbsDir); err == nil && stat.IsDir() {
+		out = os.RemoveAll(filepath.Join(checkpointDir, config.CheckpointID))
+	} else {
+		out = fmt.Errorf("checkpoint %s does not exist for container %s", config.CheckpointID, name)
```

Comment:
We may just get a permission denied here, in which case we want to print the error.